File: app/src/main/java/com/mobdeve/awitize/SignupActivity.java
package com.mobdeve.awitize;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.mobdeve.awitize.helpers.DatabaseHelper;

public class SignupActivity extends AppCompatActivity {

    private EditText email;
    private EditText password;
    private EditText cfmPassword;
    private Button btnSignUp;
    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up);

        mAuth = FirebaseAuth.getInstance();
        initComponents();
    }

    private void initComponents() {
        email = findViewById(R.id.et_email_signup);
        password = findViewById(R.id.et_password_signup);
        cfmPassword = findViewById(R.id.et_cfm_password_signup);
        btnSignUp = findViewById(R.id.bt_signup);

        btnSignUp.setOnClickListener(v -> {
            String emailInput = email.getText().toString();
            String passwordInput = password.getText().toString();
            String cfmPasswordInput = cfmPassword.getText().toString();

            if(!passwordInput.equals(cfmPasswordInput)){
                Toast.makeText(this, "Passwords are not the same", Toast.LENGTH_SHORT).show();
                return;
            }

            if(passwordInput.length() < 8){
                Toast.makeText(this, "Passwords needs a minimum of 8 characters", Toast.LENGTH_SHORT).show();
                return;
            }

            mAuth.createUserWithEmailAndPassword(emailInput, passwordInput).addOnCompleteListener(this, new OnCompleteListener<AuthResult>() {
                @Override
                public void onComplete(@NonNull Task<AuthResult> task) {

                    if(task.isSuccessful()){
                        DatabaseHelper helper = new DatabaseHelper(SignupActivity.this);
                        FirebaseUser user = task.getResult().getUser();
                        helper.addUser(user.getEmail(), user.getUid());
                        Intent i = new Intent(SignupActivity.this, DashboardActivity.class);
                        startActivity(i);
                        finish();
                    }
                    else{
                        Toast.makeText(SignupActivity.this, "Signup Not Successful", Toast.LENGTH_SHORT).show();
                        Log.w("Error", task.getException().toString());
                    }
                }
            });

        });

    }
}

File: app/src/main/java/com/mobdeve/awitize/activity/SignupActivity.kt
package com.mobdeve.awitize.activity

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.google.firebase.auth.FirebaseAuth
import com.mobdeve.awitize.R
import com.mobdeve.awitize.helpers.DatabaseHelper

class SignupActivity : AppCompatActivity() {

    private lateinit var etEmail : EditText
    private lateinit var etPassword: EditText
    private lateinit var etConfirmPassword: EditText
    private var mAuth : FirebaseAuth = FirebaseAuth.getInstance()
    private val db : DatabaseHelper = DatabaseHelper.getInstance()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)
        initComponents()
    }

    private fun initComponents(){
        etEmail = findViewById(R.id.et_signup_email)
        etPassword = findViewById(R.id.et_signup_pw)
        etConfirmPassword = findViewById(R.id.et_signup_cfm_pw)

        val btnSignup = findViewById<Button>(R.id.btn_signup_signup)
        val btnLogin = findViewById<Button>(R.id.btn_signup_login)
        btnSignup.setOnClickListener {
            val email: String = etEmail.text.toString()
            val password: String = etPassword.text.toString()
            val cfmPassword: String = etConfirmPassword.text.toString()

            if(!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()){
                Toast.makeText(this, "Invalid Email Input", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if(password.length < 8){
                Toast.makeText(this, "Password must be at least 8 characters", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            if(password != cfmPassword){
                Toast.makeText(this, "Passwords do not match", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            mAuth.createUserWithEmailAndPassword(email, password).addOnCompleteListener{
                if(it.isSuccessful){
                    db.registerUser(email, mAuth.currentUser?.uid.toString())
                    val i = Intent(this@SignupActivity, MainActivity::class.java)
                    startActivity(i)
                    finish()
                }
                else{
                    Toast.makeText(this, "Registration Failed", Toast.LENGTH_SHORT).show()
                }
            }
        }

        btnLogin.setOnClickListener {
            val intent = Intent(this, LoginActivity::class.java)
            startActivity(intent)
            finish()
        }
    }
}